LIBSEARCH-83 Initial pass at IA searcher

cfitz opened the pull request:
This is the inital code for the internet archive searcher.
Points to the advancedsearch.php endpoint.

https://archive.org/advancedsearch.php
https://issues.umd.edu/browse/LIBSEARCH-83

cfitz committed:
LIBSEARCH-83 Initial pass at IA searcher

This is the inital code for the internet archive searcher.
Points to the advancedsearch.php endpoint.

https://archive.org/advancedsearch.php
https://issues.umd.edu/browse/LIBSEARCH-83

dsteelma-umd commented:
Chris,

I checked out the "feature/LIBSEARCH-83" branch of "searchumd-special", and ran the following commands:

```
> bundle install --without production
> rails db:migrate
> rails server
```

When I ran the application, the home page came up. When I ran a search, though, I got the following error:

```
NoMethodError in QuickSearch::Search#index 
```

Do you encounter this problem when you checkout and run the searchumd-special "feature/LIBSEARCH-83" branch?

I was able to fix the error by editing the "searchumd-special/config/quick_search_config.yml" file and changing line #21 to:

```
searchers: [best_bets,archives_space,DRUM,wikipedia,internet_archive,placeholder]
```

A couple of additional comments:

1) Please fill out the "quick_search-internet_archive_searcher/README.md" file. If there documentation somewhere describing the Internet Archive search API, please provide information and a link to it.

2) When running some sample searches, I noticed that some searches don't appear to return any results. For example, if I search for "Testudo" (the name of UMD's mascot) no results are returned. I played around with the Internet Archive page at https://archive.org/details/university_maryland_cp, and noticed that beneath the search box there were "Metadata" and "Text contents" options. Searching for "Testudo" with "Metadata" selected yields no result, but using "Text contents" does.

I don't know if it is possible/feasible to set things up so that we run a "Text contents" search, but I think we should do that if we can. If it is not possible/feasible for some reason, that's okay.

cfitz committed:
LIBSEARCH-83 Update README

Added the readme file.

https://issues.umd.edu/browse/LIBSEARCH-83

cfitz commented:
Ok I added the placeholder to the searchers. Sorry didn't know that had to be there.
Updated the README. I don't think there's a way to get the API to search fulltext. I tried adding the 'sin=TXT' param to the API query but it does not see to do anything.

dsteelma-umd committed:
LIBSEARCH-83. Added "bin/rails" so Rails tests would run

Removed "bin/" from ".gitignore" file, and added "bin/rails" to
repository so that running the "rails test" command would run the
tests.

https://issues.umd.edu/browse/LIBSEARCH-83

dsteelma-umd committed:
LIBSEARCH-83. Minor copy-editing tweaks

https://issues.umd.edu/browse/LIBSEARCH-83
